src/Cache/ConfigEntry.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+namespace ConfigCat\Cache;
+
+use UnexpectedValueException;
+
+/**
+ * Represents the cached configuration.
+ * @package ConfigCat
+ */
+class ConfigEntry
+{
+    private static ?ConfigEntry $empty = null;
+
+    private function __construct(
+        private readonly string $configJson,
+        private readonly array $config,
+        private readonly string $etag,
+        private readonly int $fetchTime,
+    ) {
+    }
+
+    public function getConfigJson(): string
+    {
+        return $this->configJson;
+    }
+
+    public function getConfig(): array
+    {
+        return $this->config;
+    }
+
+    public function getEtag(): string
+    {
+        return $this->etag;
+    }
+
+    public function getFetchTime(): int
+    {
+        return $this->fetchTime;
+    }
+
+    public function serialize(): string
+    {
+        return ($this->fetchTime * 1000) . "\n" . $this->etag . "\n" . $this->configJson;
+    }
+
+    public function withTime(int $time): ConfigEntry
+    {
+        return new ConfigEntry($this->configJson, $this->config, $this->etag, $time);
+    }
+
+    public static function empty(): ConfigEntry
+    {
+        if (self::$empty == null) {
+            self::$empty = new ConfigEntry("", [], "", 0);
+        }
+
+        return self::$empty;
+    }
+
+    public static function fromConfigJson(string $configJson, string $etag, int $fetchTime): ConfigEntry
+    {
+        $deserialized = json_decode($configJson, true);
+        if ($deserialized == null) {
+            return self::empty();
+        }
+
+        return new ConfigEntry($configJson, $deserialized, $etag, $fetchTime);
+    }
+
+    public static function fromCached(string $cached): ConfigEntry
+    {
+        $timePos = strpos($cached, "\n");
+        $etagPos = strpos($cached, "\n", $timePos + 1);
+
+        if ($timePos === false || $etagPos === false) {
+            throw new UnexpectedValueException("Number of values is fewer than expected.");
+        }
+
+        $fetchTimeString = substr($cached, 0, $timePos);
+        $fetchTime = intval($fetchTimeString);
```

Comment:
Won't be this conversion problematic on 32-bit systems? The [PHP manual](https://www.php.net/manual/en/function.intval.php) says:

> 32 bit systems have a maximum signed integer range of -2147483648 to 2147483647. So for example on such a system, intval('1000000000000') will return 2147483647. The maximum signed integer value for 64 bit systems is 9223372036854775807.

A floating point number looks safer to represent fetch time.

Reply:
It won't be for the next 15 years 😄 

Anyway, I changed it to float!

Reply:
I'm afraid that would be true if the cache entry stored the fetch time as seconds. But, we made that to store it as milliseconds, so we have problems right now...